include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,220 @@ struct video_frmival_enum {
 	};
 };
 
+/**
+ * @name Port and Endpoint DT Helpers
+ * @{
+ */
+
+#define _DT_INST_PORT_BY_ID(inst, pid)                                                             \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(inst, ports)),                                    \
+	    (DT_CHILD(DT_INST_CHILD(inst, ports), port_##pid)), (DT_INST_CHILD(inst, port_##pid)))
+
+/**
+ * @brief Get a port node from its id
+ *
+ * Given a device instance number, return a port node specified by its ID.
+ * It handles various ways of how a port could be defined. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_INST_PORT_BY_ID(inst, 0), property)
+ * @endcode
+ *
+ * This above snippet will return @c "port_property" for each @c &device listed below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		ports {
+ *			port@0 {
+ *				reg = <0x0>;
+ *				property = "port_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port@0 {
+ *			reg = <0x0>;
+ *			property = "port_property";
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			property = "port_property";
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @return port node associated with @p pid
+ */
+#define DT_INST_PORT_BY_ID(inst, pid)                                                              \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(inst, pid)),                                \
+		(_DT_INST_PORT_BY_ID(inst, pid)), (DT_INST_CHILD(inst, port)))
+
+#define _DT_INST_ENDPOINT_BY_ID(inst, pid, eid)                                                    \
+	DT_CHILD(DT_INST_PORT_BY_ID(inst, pid), endpoint_##eid)
+
+/**
+ * @brief Get an endpoint node from its id and its parent port id
+ *
+ * Given a device instance number, a port ID and an endpoint ID, return the endpoint node.
+ * It handles various ways of how a port and an endpoint could be defined as described in
+ * @ref DT_INST_PORT_BY_ID and below. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_INST_ENDPOINT_BY_ID(inst, 0, 0), property)
+ * @endcode
+ *
+ * The above snippet returns @c "endpoint_property" for each of the @c &device listed below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			endpoint {
+ *				property = "endpoint_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			endpoint@0 {
+ *				reg = <0x0>;
+ *				property = "endpoint_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		ports {
+ *			port@0 {
```

Comment:
Note: I did not include the `#size-cells = <0>;` and `#address-cells = <1>` here for compactness.
No idea if it is best to add that or not.

Reply:
yes, I think we should. Whenever there is multiple ports or endpoints, the parent node need to declare `#size-cells` and `#address-cells`

Reply:
done